### ISSUE TITLE
torchpack stalls on startup

### ISSUE THREAD

**Issue by martinrebane** [zhijian-liu/torchpack#31]
Hi! Strange problem with torchpack as it stalls on startup. I am evaluating SPVCNN models and when I launch the script it sometimes starts, but sometimes such stalls and does nothing.

```
torchpack dist-run -np 1 python evaluate.py configs.yaml --name run-1fb33cb7
```
There are no errors, `python evaluate.py` process is started but there is no CPU or GPU load. I will then hit ctrl+C and start the same command again and it will run eventually (at least after a few tries). But it is very annoying. Some kind of deadlock? Do you have any suggestions on how to debug this and find the cause? Are there any debug parameters for torchpack?

**Comment by zhijian-liu** [comment]
Thanks for reporting this issue. You could add `-v` to the training command to see whether there is any error or warning:
```
torchpack dist-run -np 1 -v python evaluate.py configs.yaml --name run-1fb33cb7
```

**Comment by martinrebane** [comment]
Hi! Thanks for the tip! -v did print out mpi-run command, but there are neither errors nor warnings. The last thing that is prints out before stalling is "Failed to import tensorflow." This is normal as I do not have Tensorflow installed, but indicates where the problem occurs. When the run is successful, then the next line will be
```
[2022-01-25 16:25:22.244] /opt/conda/bin/python evaluate.py configs.yaml --name run-1fb33cb7
[2022-01-25 16:25:22.244] Experiment started: ...
```
So it stalls inbetween without errors and warnings, also no CPU and GPU activity. Some kind of deadlock? Perhaps caused by config.yaml (it is from training having training_size and workers_per_gpu from training setup). Should I adjust them for evaluation?

**Comment by martinrebane** [comment]
No idea if it matters but when I observe process list using htop then both have a main `mpirun` process with 3 `mpirun` subprocess and 1 `python` subprocess like this:
![Screenshot from 2022-01-25 19-05-02](https://user-images.githubusercontent.com/11771536/151024657-22f801f2-9982-4004-a4c1-79fa89f3e684.png)
But while the successful run has a huge nested process tree:
![Screenshot from 2022-01-25 18-49-12](https://user-images.githubusercontent.com/11771536/151024729-8884b9f7-25b8-4457-bafe-af19d8a3e8e2.png)
... stalled run has only a few:
![Screenshot from 2022-01-25 18-47-22](https://user-images.githubusercontent.com/11771536/151024781-1b9d8182-333f-42ad-b41b-9e98bdba3eb2.png)

**Comment by zhijian-liu** [comment]
That's very interesting. Could you try to comment all the code and just leave `dist.init()` there to see whether you can still reproduce this error? Thanks!

**Comment by martinrebane** [comment]
That was a good tip! I have narrowed down the problem to `myavi` import statement. Here is the code that freezes `dist.init()`:
```
from torchpack import distributed as dist

import mayavi.mlab as mlab

def main() -> None:
    dist.init()
    print("dist init ok")

if __name__ == '__main__':
    main()
```
When I remove `mayavi` import then all works fine. Without it, the code freezes on dist.init(). Every 3rd or 5th run succeeds and prints out `dist init ok`, other runs just freeze after printing out `mpirun` command and never reach `print("dist init ok")`

Python 3.7.10, torch 1.8.0+cu111,  torchpack 0.3.1

Clearly, the rather random freezing of `dist.init()` is caused by `import mayavi.mlab as mlab` but why and is there something to do about it? I also tried running
```
xvfb-run --server-args="-screen 0 1024x768x24" torchpack dist-run -np 1 -v python dist.py
```
But it is the same, still freezes on `dist.init()`.
